Sat Jan 15 20:30:46 CET 2005  Norbert@Nemec-online.de
  * Py_DECREF
Sat Jan 15 16:11:15 CET 2005  Norbert@Nemec-online.de
  * _POSIX_C_SOURCE
Index: python/Include/object.h
===================================================================
--- python.orig/Include/object.h	2006-04-12 19:06:58.000000000 +0200
+++ python/Include/object.h	2006-11-14 12:37:12.000000000 +0100
@@ -615,12 +615,14 @@
 	_Py_INC_REFTOTAL  _Py_REF_DEBUG_COMMA	\
 	(op)->ob_refcnt++)
 
-#define Py_DECREF(op)					\
-	if (_Py_DEC_REFTOTAL  _Py_REF_DEBUG_COMMA	\
-	    --(op)->ob_refcnt != 0)			\
-		_Py_CHECK_REFCNT(op)			\
-	else						\
-		_Py_Dealloc((PyObject *)(op))
+#define Py_DECREF(op)						\
+	do {							\
+		if (_Py_DEC_REFTOTAL  _Py_REF_DEBUG_COMMA	\
+		    --(op)->ob_refcnt != 0)			\
+			_Py_CHECK_REFCNT(op)			\
+		else						\
+			_Py_Dealloc((PyObject *)(op));		\
+	} while(0)
 
 #define Py_CLEAR(op)				\
         do {                            	\
Index: python/Include/Python.h
===================================================================
--- python.orig/Include/Python.h	2004-07-27 17:57:24.000000000 +0200
+++ python/Include/Python.h	2006-11-14 12:37:12.000000000 +0100
@@ -5,6 +5,7 @@
 /* Include nearly all Python header files */
 
 #include "patchlevel.h"
+#undef _POSIX_C_SOURCE
 #include "pyconfig.h"
 
 /* Cyclic gc is always enabled, starting with release 2.3a1.  Supply the
